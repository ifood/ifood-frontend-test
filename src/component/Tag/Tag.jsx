import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Button, Label } from 'semantic-ui-react';
import styled from 'styled-components';

const TagWrapper = styled.div`
  flex-shrink: 0;
  margin: 5px;
`;

const Tag = ({ id, field, value, onDelete }) => {
  function handleDelete() {
    if (onDelete != null) {
      onDelete(id);
    }
  }

  return (
    <TagWrapper>
      <Button as="div" labelPosition="left">
        <Label basic>
          {field}: {value}
        </Label>
        <Button onClick={handleDelete} icon>
          <Icon name="remove" />
        </Button>
      </Button>
    </TagWrapper>
  );
};

Tag.propTypes = {
  id: PropTypes.string,
  field: PropTypes.string,
  value: PropTypes.string,
  onDelete: PropTypes.func,
};

export default Tag;
