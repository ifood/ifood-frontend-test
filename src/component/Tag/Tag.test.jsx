import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tag from './Tag';

describe('<Tag />', () => {
  it('snapshot', () => {
    const wrapper = mount(<Tag field="field" value="value" />);
    jestExpect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call onDelete (with id as param) when delete button is clicled', () => {
    const onDelete = jest.fn();
    const id = 'tagId';
    const tag = mount(<Tag id={id} onDelete={onDelete} />);
    tag.find('button').simulate('click');
    expect(onDelete.mock.calls.length).to.be.equal(1);
    expect(onDelete.mock.calls[0][0]).to.be.equal(id);
  });

  it("should not throw an error if onDelete wasn't provided", () => {
    const tag = mount(<Tag />);
    function click() {
      tag.find('button').simulate('click');
    }
    expect(click).to.not.throw();
  });
});
