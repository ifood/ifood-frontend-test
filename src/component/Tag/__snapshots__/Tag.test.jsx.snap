// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> snapshot 1`] = `
.c0 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin: 5px;
}

<Tag
  field="field"
  value="value"
>
  <styled.div>
    <div
      className="c0"
    >
      <Button
        as="div"
        labelPosition="left"
      >
        <Ref
          innerRef={
            Object {
              "current": <div
                class="ui left labeled button"
                role="button"
                tabindex="0"
              >
                <div
                  class="ui basic label"
                >
                  field
                  : 
                  value
                </div>
                <button
                  class="ui icon button"
                >
                  <i
                    aria-hidden="true"
                    class="remove icon"
                  />
                </button>
              </div>,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <div
                  class="ui left labeled button"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="ui basic label"
                  >
                    field
                    : 
                    value
                  </div>
                  <button
                    class="ui icon button"
                  >
                    <i
                      aria-hidden="true"
                      class="remove icon"
                    />
                  </button>
                </div>,
              }
            }
          >
            <div
              className="ui left labeled button"
              onClick={[Function]}
              role="button"
              tabIndex={0}
            >
              <Label
                basic={true}
              >
                <div
                  className="ui basic label"
                  onClick={[Function]}
                >
                  field
                  : 
                  value
                </div>
              </Label>
              <Button
                as="button"
                icon={true}
                onClick={[Function]}
              >
                <Ref
                  innerRef={
                    Object {
                      "current": <button
                        class="ui icon button"
                      >
                        <i
                          aria-hidden="true"
                          class="remove icon"
                        />
                      </button>,
                    }
                  }
                >
                  <RefFindNode
                    innerRef={
                      Object {
                        "current": <button
                          class="ui icon button"
                        >
                          <i
                            aria-hidden="true"
                            class="remove icon"
                          />
                        </button>,
                      }
                    }
                  >
                    <button
                      className="ui icon button"
                      onClick={[Function]}
                    >
                      <Icon
                        as="i"
                        name="remove"
                      >
                        <i
                          aria-hidden="true"
                          className="remove icon"
                          onClick={[Function]}
                        />
                      </Icon>
                    </button>
                  </RefFindNode>
                </Ref>
              </Button>
            </div>
          </RefFindNode>
        </Ref>
      </Button>
    </div>
  </styled.div>
</Tag>
`;
